### PR TITLE
[firmware transparency] Use `formats/log` checkpoint format

### PR DESCRIPTION
**Pull request by AlCutter**
Best reviewed commit-by-commit.

### CODE REVIEW

File: binary_transparency/firmware/internal/verify/bundle.go [hunk 1]
```diff
@@ -44,42 +45,57 @@ func BundleForUpdate(bundleRaw, fwHash []byte, dc api.LogCheckpoint, cpFunc Cons
...
+		if err := lv.VerifyConsistencyProof(int64(dc.Size), int64(pc.Size), dc.Hash, pc.Hash, cProof); err != nil {
 			return proofBundle, fwMeta, fmt.Errorf("failed verification of consistency proof %w", err)
 		}
 	}
 	return proofBundle, fwMeta, nil
 }
 
 // BundleConsistency verifies the log checkpoint in the bundle is consistent against a given checkpoint (e.g. one fetched from a witness).
-func BundleConsistency(pb api.ProofBundle, rc api.LogCheckpoint, cpFunc ConsistencyProofFunc) error {
+func BundleConsistency(pb api.ProofBundle, rc api.LogCheckpoint, cpFunc ConsistencyProofFunc, cpSigVerifier note.Verifier) error {
```
Comment (mhutchinson) [inline binary_transparency/firmware/internal/verify/bundle.go:73]: cpSigVerifier in this specific context easily leads to expanding cp to "consistency proof". I wonder if `c` or `checkpoint` or `log` would be better prefixes?

**Review by mhutchinson — APPROVED** [review]
Fair play for getting through all of that!

File: binary_transparency/firmware/internal/client/client.go [hunk 2]
```diff
@@ -131,18 +134,28 @@ func (c ReadonlyClient) GetCheckpoint() (*api.LogCheckpoint, error) {
 		return &api.LogCheckpoint{}, errFromResponse("failed to fetch checkpoint", r)
 	}
 
-	var cp api.LogCheckpoint
-	if err := json.NewDecoder(r.Body).Decode(&cp); err != nil {
+	b, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read body: %w", err)
+	}
+	defer r.Body.Close()
```
[the remark this answers is not in the document]
Comment (AlCutter) [reply binary_transparency/firmware/internal/client/client.go:141]: Good spot - this should've been up right after we know that the `Get` was successful really - moved up there.

File: binary_transparency/firmware/cmd/ft_witness/ft_witness.go [hunk 3]
```diff
@@ -22,6 +22,8 @@ import (
 
 	"github.com/golang/glog"
 	"github.com/google/trillian-examples/binary_transparency/firmware/cmd/ft_witness/impl"
+	"github.com/google/trillian-examples/binary_transparency/firmware/internal/crypto"
+	"golang.org/x/mod/sumdb/note"
```
[the remark this answers is not in the document]
Comment (AlCutter) [reply binary_transparency/firmware/cmd/ft_witness/ft_witness.go:26]: Not that I'm aware of, but we can always fork it if necessary?

File: binary_transparency/firmware/internal/client/wclient.go [hunk 4]
```diff
@@ -41,15 +42,23 @@ func (c WitnessClient) GetWitnessCheckpoint() (*api.LogCheckpoint, error) {
...
 	}
+	defer r.Body.Close()
 	if r.StatusCode != 200 {
-		return &api.LogCheckpoint{}, errFromRsp("failed to fetch checkpoint", r)
+		return nil, errFromRsp("failed to fetch checkpoint", r)
 	}
 
-	var wcp api.LogCheckpoint
-	if err := json.NewDecoder(r.Body).Decode(&wcp); err != nil {
+	b, err := ioutil.ReadAll(r.Body)
```
[the remark this answers is not in the document]
Comment (AlCutter) [reply binary_transparency/firmware/internal/client/wclient.go:50]: I think this is already done at L45?

File: binary_transparency/firmware/cmd/ft_witness/ft_witness.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mhutchinson) [reply binary_transparency/firmware/cmd/ft_witness/ft_witness.go:26]: Yeah, that's a reasonable option if we need/want to shape the dependency graph.

File: binary_transparency/firmware/internal/verify/bundle.go (hunk 1 above)
Comment (AlCutter) [reply binary_transparency/firmware/internal/verify/bundle.go:73]: Yeah, I think I'll move to `logSigVerifier` everywhere

File: binary_transparency/firmware/internal/client/wclient.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mhutchinson) [reply binary_transparency/firmware/internal/client/wclient.go:50]: So it is!

File: binary_transparency/firmware/cmd/ft_personality/ft_personality.go [hunk 5]
```diff
@@ -40,13 +42,19 @@ var (
 func main() {
 	flag.Parse()
 
+	signer, err := note.NewSigner(crypto.TestFTPersonalityPriv)
+	if err != nil {
+		panic(err)
```
[the remark this answers is not in the document]
Comment (AlCutter) [reply binary_transparency/firmware/cmd/ft_personality/ft_personality.go:47]: Done

**Pull request merged**